runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -113,12 +115,69 @@ func (q *MetricsViewComparison) Resolve(ctx context.Context, rt *runtime.Runtime
 	}
 
 	if !isTimeRangeNil(q.ComparisonTimeRange) {
+		// execute toplist for base and get dim list
+		// create and add filter and execute comprison toplist
+		// remove strict limits in comp toplist sql
+		if drivers.DialectDruid != olap.Dialect() || q.NoExploreExactifyMode {
+			return q.executeComparisonToplist(ctx, olap, mv, priority, security)
+		}
+
+		if q.Exact {
+			return fmt.Errorf("Cannot calculate Exact when Druid Exactify is requested")
+		}
+
+		// Druid-based `exactify` approach:
+		// The previous query collected a list of dimensions. The following query will filter in those dimensions
+		// to reduce the topN list to fit in memory to exclude discrepancies due incomplete topN tables merged by the previous query
+		if q.isBase() || q.isDeltaComparison() {
+			err = q.executeToplist(ctx, olap, mv, priority, security)
+			if err != nil {
+				return err
+			}
+		} else {
+			ttr := q.TimeRange
+			q.TimeRange = q.ComparisonTimeRange
+			err = q.executeToplist(ctx, olap, mv, priority, security)
+			if err != nil {
+				return err
+			}
+
+			q.TimeRange = ttr
+		}
+		q.addDimsAsFilter()
```

Comment:
We need not execute the query again if the prev query had all the measures already computes and gives us less results  than the limit...
e.g.  e.g cardinality is < 10 and we asked for Top100 then our first query already gave exact results..